### ISSUE TITLE
Update assembly parsers to use u32assert2 operation where possible

### ISSUE THREAD

**Issue by bobbinth** [maticnetwork/miden#216]
Currently, for most safe u32 assembly instructions we check that u32 operands are u32 values by using an equivalent of `u32assert` instruction (which actually takes 3 cycles). However, now we have a `U32ASSERT2` operation which could make all of these checks more efficient.

For example, `u32add` assembly instruction currently translates into something like:
```
PAD U32ASSERT2 DROP
SWAP
PAD U32ASSERT2 DROP
U32ADD
EQZ
ASSERT
```
This takes 10 cycles. Instead, using `U32ASSERT2` operation, we could do something like this:
```
U32ASSERT2
U32ADD
EQZ
ASSERT
```
This would take just 4 cycles.

Similar logic would apply to most other "safe" instructions.

**Comment by 0xKanekiKen** [comment]
I'm compiling a list of assembly operations which needs to be updated with `U32ASSERT2`. I'll post them here.

**Comment by DeimosTheMisthios** [comment]
Thanks Kaneki, I'd love to pick this up when that list is ready.

**Comment by 0xKanekiKen** [comment]
Thanks @DeimosTheMisthios, for the help. I will compile it in a day or two & then you can take it up from there.

**Comment by 0xKanekiKen** [comment]
The following list of assembly instructions needs to be updated:

- [ ] `u32add`
- [ ] `u32add.b`
- [ ] `u32add.full`
- [ ] `u32sub`
- [ ] `u32sub.b`
- [ ] `u32sub.full`
- [ ] `u32mul`
- [ ] `u32mul.b`
- [ ] `u32mul.full`
- [ ] `u32madd`
- [ ] `u32div`
- [ ] `u32div.b`
- [ ] `u32div.full`
- [ ] `u32eq`
- [ ] `u32mod`
- [ ] `u32shl`
- [ ] `u32shr`
- [ ] `u32lt`
- [ ] `u32lte`
- [ ] `u32gt`
- [ ] `u32gte`
- [ ] `u32min`
- [ ] `u32max`

I've removed `u32add3` & `u32assertw` from the above list as they already had been actioned. 

P.S. - Almost all the above opcodes are using `handle_arithmetic_operation`& `assert_u32_operands` helper functions to handle u32 assertion. Updating them would fix almost all of them at one go.

**Comment by tnachen** [comment]
Seems quite straightforward, I can work on this as looks like @DeimosTheMisthios you are working on other issues.

**Comment by bobbinth** [comment]
Closed by #228
